### ISSUE TITLE
E0403 is not detected then associated item shadows parameters from the containing item

### ISSUE THREAD

**Issue by neonaot** [intellij-rust/intellij-rust#9949]
## Environment

* **IntelliJ Rust plugin version:** 0.4.186.5128-223-beta
* **Rust toolchain version:** 1.68.0-nightly (388538fc9 2023-01-05) x86_64-apple-darwin
* **IDE name and version:** IntelliJ IDEA 2022.3.1 Ultimate Edition (IU-223.8214.52)
* **Operating system:** macOS 12.5.1
* **Macro expansion:** enabled

## Problem description
See https://doc.rust-lang.org/error_codes/E0403.html
<img width="852" alt="image" src="https://user-images.githubusercontent.com/49211026/211294847-cf9992ce-f2aa-453b-b3e5-9f2a36b372df.png">
## Steps to reproduce
```rust

trait Foo<T> {
        fn do_something(&self) -> T;
        fn do_something_else<T: Clone>(&self, bar: T);
}

```
